src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -486,6 +491,17 @@ private DRSAccessURL getAccessURL(
 
     BillingProfileModel billingProfileModel = cachedSnapshot.datasetBillingProfileModel;
 
+    loggingMetrics.set(BardEventProperties.DATASET_ID_FIELD_NAME, cachedSnapshot.getDatasetId());
+    loggingMetrics.set(
+        BardEventProperties.DATASET_NAME_FIELD_NAME, cachedSnapshot.getDatasetName());
+    loggingMetrics.set(BardEventProperties.SNAPSHOT_ID_FIELD_NAME, snapshotId);
+    loggingMetrics.set(BardEventProperties.SNAPSHOT_NAME_FIELD_NAME, cachedSnapshot.getName());
+    loggingMetrics.set(
+        BardEventProperties.BILLING_PROFILE_ID_FIELD_NAME,
+        cachedSnapshot.getSnapshotBillingProfileId());
+    loggingMetrics.set(
+        BardEventProperties.CLOUD_PLATFORM_FIELD_NAME, cachedSnapshot.getCloudPlatform());
+
```

Comment:
This block feels to me like it would be a good fit in a helper method with a docstring, which would make it easier to expand in the future if needed.  Your helper method could get away with just taking `cachedSnapshot` as an input.